server/http.go

```diff
@@ -135,6 +140,59 @@ func (s *HTTPd) RegisterSiteHandlers(site site.API, auth auth.Auth, cache *util.
 		"deletedevice": {"DELETE", "/devices/{class:[a-z]+}/{id:[0-9.]+}", deleteDeviceHandler},
 		"testconfig":   {"POST", "/test/{class:[a-z]+}", testConfigHandler},
 		"testmerged":   {"POST", "/test/{class:[a-z]+}/merge/{id:[0-9.]+}", testConfigHandler},
+
+		// new endpoints ⤵︎
+		// "hems":       {"GET", "/hems", settingsGetStringHandler("hems")},
+		// "updatehems": {"POST", "/hems", settingsSetYamlHandler("hems", new(config.Typed))},
+		// "deletehems": {"DELETE", "/hems", settingsDeleteHandler("hems")},
+		// "tariffs":           {"GET", "/tariffs", settingsGetStringHandler("tariffs")},
+		// "updatetariffs":     {"POST", "/tariffs", settingsSetYamlHandler("tariffs", new(globalconfig.Tariffs))},
+		// "deletetariffs":     {"DELETE", "/tariffs", settingsDeleteHandler("tariffs")},
+		// "messaging":         {"GET", "/messaging", settingsGetStringHandler("messaging")},
+		// "updatemessaging":   {"POST", "/messaging", settingsSetYamlHandler("messaging", new(globalconfig.Messaging))},
+		// "deletemessaging":   {"DELETE", "/messaging", settingsDeleteHandler("messaging")},
+		// "modbusproxy":       {"GET", "/modbusproxy", settingsGetStringHandler("modbusproxy")},
+		// "updatemodbusproxy": {"POST", "/modbusproxy", settingsSetYamlHandler("modbusproxy", new(globalconfig.ModbusProxy))},
+		// "deletemodbusproxy": {"DELETE", "/modbusproxy", settingsDeleteHandler("modbusproxy")},
+		// "mqtt":               {"GET", "/mqtt", settingsGetJsonHandler("mqtt", new(globalconfig.Mqtt))},
+		// "updatemqtt":         {"POST", "/mqtt", settingsSetJsonHandler("mqtt", new(globalconfig.Mqtt))},
+		// "deletemqtt":         {"DELETE", "/mqtt", settingsDeleteHandler("mqtt")},
+		// "influx":             {"GET", "/influx", settingsGetJsonHandler("influx", new(globalconfig.Influx))},
+		// "updateinflux":       {"POST", "/influx", settingsSetJsonHandler("influx", new(globalconfig.Influx))},
+		// "deleteinflux":       {"DELETE", "/influx", settingsDeleteHandler("influx")},
+		// "eebus":              {"GET", "/eebus", settingsGetJsonHandler("eebus", new(eebus.Config))},
+		// "updateeebus":        {"POST", "/eebus", settingsSetJsonHandler("eebus", new(eebus.Config))},
+		// "deleteeebus":        {"DELETE", "/eebus", settingsDeleteHandler("eebus")},
+		// "network":            {"GET", "/network", settingsGetJsonHandler("network", new(globalconfig.Network))},
+		// "updatenetwork":      {"POST", "/network", settingsSetJsonHandler("network", new(globalconfig.Network))},
+		// "deletenetwork":      {"DELETE", "/network", settingsDeleteHandler("network")},
+		"interval":           {"POST", "/interval/{value:[0-9.]+", settingsSetDurationHandler("interval")},
+		"updatesponsortoken": {"POST", "/sponsortoken/{token:[a-zA-Z0-9_-.]+}", updateSponsortokenHandler},
+		"deletesponsortoken": {"DELETE", "/sponsortoken", settingsDeleteHandler(keys.SponsorToken)},
```

Comment:
Hier würd ich lieber nen dedizierten Endpunkt haben, der auch den aktualisierten Sponsor-Status korrekt neu publisht. Im Gegensatz zu den anderen Feldern find ich es hier sehr wichtig, dass der Nutzer eine direkte Bestätigung/Validierung in der UI bekommt. Also ohne Neustart.

Reply:
Räums gerne auf- der letzte ist bewusst noch drin. Wir brauchen wohl 2 Delete Handler- empty und leeres JSON.

Reply:
Ich hab den handler nun so umgebaut, dass delete immer nen leeren String setzt. Dafür kann der getJson handler damit jetzt umgehen und liefert den default struct zurück. vorher kam stattdessen immer ein leeres `{}` wenn einmal DELETEed wurde.